roadmap/implementers-guide/src/runtime/disputes.md

```diff
@@ -27,6 +27,8 @@ This is necessary to slash the losing party appropriately which is based on the
 
 ## Local Disputes
 
+Who does local disputes, validators self-select based on a [VRF](##-Secondary-Approval-Checking).
```

Comment:
A dispute is still a dispute, so the VRFs no longer matter.  Any successful dispute creates a remote dispute so the chain that survives still slashed the wrong voters.

Reply:
> A dispute is still a dispute, so the VRFs no longer matter.

Mostly meant to keep track of what is triggering this whole process.

> Any successful dispute creates a remote dispute so the chain that survives still slashed the wrong voters, so maybe local disputes should be handled as closely as possible to remote disputes.

So far I was viewing them as kind of distinct things, for a local dispute one does not have to care about being fed an initial dispute trigger for a invalid/non-existant/non-available block.

> Imagine we've some `LocalDisputeMessage` and `RemoteDisputeMessage` that achieve the same thing in dispute processing. Is there any reason in having `LocalDisputeMessage`? I suspect no because `LocalDisputeMessage` being smaller or faster does not warrant extra code. It's possible you'll find some chain condition that warrants special checking for the local chain, but maybe the only such condition is "kill the current chain because a local dispute passed", but that's kinda off-chain, no?

Frankly, I am still not sure about this part. It's a call to an on-chain API, so it would make sense to have an inherent as part of the transaction that would trigger blacklisting/rollback to a certain block. Calling on chain API from off-chain is possible, but nothing is persisted iirc.

Reply:
I donno all the details about telling a chain to halt and fork, but.. It must influence the fork-choice rule elsewhere, so maybe it could live almost entirely in the fork-choice rule, which is off-chain, but not sure.. like maybe you need an on-chain component simply for robust error handling.

Reply:
Yeah, I figure it'll all live in the fork-choice rule.

Reply:
> A dispute is still a dispute, so the VRFs no longer matter.

So the secondary checkers are the secondary checkers of the dispute or the secondary checkers of the approval process?

Reply:
If we have a dispute then it no longer matters who were the approval (secondary) checkers.  

Imagine we've forks B1,B2 > B0 that contain the same candidate C then B1 and B2 imply different approval checkers for C.  If one validator, not necessarily even an approval checker anywhere, disputes C, then we'll wind up with another fork B3 > B0 that remote disputes C in both B1 and B2.

In B3+ (B3 and children), we post both the backing votes on C from B0- (B0 and ancestors) and the votes from approval checkers for C in B1+ and B2+.  Yet, all the dispute in B3+ cares about is votes, not the conditions that made those votes relevant in B1+ and B2+.  In fact, we never reveal in B3+ that B1 or B2 even exist, nor who were the approval checkers there.